debian/bionic/foreman/foreman-plugin.mk

```diff
@@ -3,17 +3,19 @@ build:
 	cp $(PLUGIN).rb /usr/share/foreman/bundler.d/
 	cd /usr/share/foreman && ( \
 		/usr/bin/foreman-ruby /usr/bin/bundle install --local && \
-		/usr/bin/npm install --no-audit --no-optional --unsafe-perm && \
-		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:assets:precompile[$(PLUGIN)] RAILS_ENV=production DATABASE_URL=nulldb://nohost && \
-		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:apipie:cache[$(PLUGIN)] cache_part=resources \
-			OUT=/var/lib/foreman/public/apipie-cache/plugin/$(PLUGIN) RAILS_ENV=development \
-			DATABASE_URL=nulldb://nohost FOREMAN_APIPIE_LANGS=en_US \
+		if [ -n "$(PLUGIN_HAS_PACKAGE_JSON)" ]; then /usr/bin/npm install --no-audit --no-optional --unsafe-perm; fi && \
+		if [ -n "$(PLUGIN_HAS_ASSETS)" ] || [ -n "$(PLUGIN_HAS_WEBPACK)" ]; then /usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:assets:precompile[$(PLUGIN)] RAILS_ENV=production DATABASE_URL=nulldb://nohost; fi && \
+		if [ -n "$(PLUGIN_HAS_API)" ]; then \
+			/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:apipie:cache[$(PLUGIN)] cache_part=resources \
+				OUT=/var/lib/foreman/public/apipie-cache/plugin/$(PLUGIN) RAILS_ENV=development \
+				DATABASE_URL=nulldb://nohost FOREMAN_APIPIE_LANGS=en_US; \
+		fi \
 	)
 	GEM_PATH=$$(cd /usr/share/foreman && /usr/bin/foreman-ruby /usr/bin/bundle show $(PLUGIN)) && \
```

Comment:
I notice this now and unrelated to this PR, but `bundle show` started to output that it's deprecated and `bundle info` should be used. That's not great because `bundle show` is exactly what we need while info has more.